upgrade/upgrade.go

```diff
@@ -39,92 +42,115 @@ func CreateUpgradeHandler(
 	}
 
-	if err := dollarKeeper.Principal.Clear(ctx, nil); err != nil {
-		return errors.Wrap(err, "unable to clear all dollar principal")
-	}
-	user, err := addressCodec.StringToBytes("noble1exg6r3tz2tup8ewnmttkkhd7qfx39rguzqngyc")
+	denom := dollarKeeper.GetDenom()
+	coins := sdk.NewCoins(bankKeeper.GetBalance(ctx, account, denom))
+
+	err = bankKeeper.SendCoinsFromAccountToModule(ctx, account, dollartypes.ModuleName, coins)
 	if err != nil {
-		return errors.Wrap(err, "unable to decode user address")
+		return errors.Wrap(err, "failed to transfer usdn to dollar module")
 	}
-	if err := dollarKeeper.Principal.Set(ctx, user, math.NewInt(964914)); err != nil {
-		return errors.Wrap(err, "unable to set dollar principal")
+	err = bankKeeper.BurnCoins(ctx, dollartypes.ModuleName, coins)
+	if err != nil {
+		return errors.Wrap(err, "unable to burn usdn from dollar module")
 	}
 
-	stats := dollartypes.Stats{
-		TotalHolders:      1,
-		TotalPrincipal:    math.NewInt(964914),
-		TotalYieldAccrued: math.NewInt(1411),
+	supply := bankKeeper.GetSupply(ctx, dollarKeeper.GetDenom())
+	if !supply.IsZero() {
+		return fmt.Errorf("expected no usdn supply, got %s", supply.Amount)
 	}
-	if err := dollarKeeper.Stats.Set(ctx, stats); err != nil {
-		return errors.Wrap(err, "unable to set dollar stats")
+
+	return nil
+}
+
+// ClearDollarModuleState clears the entire key-value store of the Noble Dollar module.
+func ClearDollarModuleState(ctx sdk.Context, dollarKey *storetypes.KVStoreKey) error {
+	dollarStore := ctx.KVStore(dollarKey)
+	iterator := dollarStore.Iterator(nil, nil)
+
+	for ; iterator.Valid(); iterator.Next() {
+		dollarStore.Delete(iterator.Key())
 	}
 
-	vaultsStats := dollarvaultstypes.Stats{
-		FlexibleTotalPrincipal:                   math.ZeroInt(),
-		FlexibleTotalUsers:                       0,
-		FlexibleTotalDistributedRewardsPrincipal: math.ZeroInt(),
-		StakedTotalPrincipal:                     math.ZeroInt(),
-		StakedTotalUsers:                         0,
+	return iterator.Close()
+}
+
+// ConfigureDollarPortalState sets the Noble Dollar Portal owner.
+func ConfigureDollarPortalState(ctx context.Context, dollarKeeper *dollarkeeper.Keeper) (err error) {
+	err = dollarKeeper.PortalOwner.Set(ctx, "noble1mx48c5tv6ss9k7793n3a7sv48nfjllhxkd6tq3")
+	if err != nil {
+		return errors.Wrap(err, "unable to set dollar portal owner in state")
 	}
-	if err := dollarKeeper.VaultsStats.Set(ctx, vaultsStats); err != nil {
-		return errors.Wrap(err, "unable to set dollar vaults stats")
+
+	return nil
+}
+
+// ConfigureWormholeState sets the Wormhole configuration.
+func ConfigureWormholeState(ctx context.Context, wormholeKeeper *wormholekeeper.Keeper) (err error) {
+	err = wormholeKeeper.Config.Set(ctx, wormholetypes.Config{
+		ChainId:          4009,
+		GuardianSetIndex: 0,
+		GovChain:         1,
+		GovAddress:       common.FromHex("0x0000000000000000000000000000000000000000000000000000000000000004"),
```

Comment:
also don't know where to cross check this address but going to assume it is correct

Reply:
This can be verified via the Wormhole SDK [here](https://github.com/wormhole-foundation/wormhole/blob/3797ed082150e6d66c0dce3fea7f2848364af7d5/sdk/vaa/governance.go#L11)!